README.md

```diff
@@ -6,10 +6,120 @@ go-dagaggregator-unixfs
+
+### Preamble
+
+This is always the first record of the manifest, it signals how to parse the
+rest of the manifest.
+```
+{
+  "RecordType": "DagAggregatePreamble",
+  "Version": 1
+}
+```
+
+### Summary
+
+This is the second record within the manifest. Has entry count and various other
+metadata.
+```
+{
+  "RecordType": "DagAggregateSummary",
+  "EntryCount": 4,
+  "EntriesSortedBy": "DagCidV1",
+  "Description": "Aggregate of non-related DAGs, produced by github.com/filecoin-project/go-dagaggregator-unixfs"
+}
+```
+
+### Individual DAG Entries
+
+The rest of the manifest records contain information about the incuded DAGs, one
+per record.
+
+```
+{
+  "RecordType": "DagAggregateEntry",
+  "DagCidV1": "bafybeibhbx3y6tnn7q4gpsous6apnobft5jybvroiepdsmvps2lmycjjxu",
+  "DagCidV0": "QmQy6xmJhrcC5QLboAcGFcAE1tC8CrwDVkrHdEYJkLscrQ",
+  "DagSize": 42,
+  "NodeCount": 1,
+  "PathPrefixes": [ "baf...xu", "baf...jjxu" ],
+  "PathIndexes": [ 2, 0, 0 ]
+}
+```
+
+`PathPrefixes` contains the parent directories for this particular DAG, absed on the chosen parts of the target CID. They currently can be derived from `DagCIDV1`, but are included for ease of navigation.
+
+`PathIndexes` contains the 0-based position of each entry within its
+corresponding parent directory. It is provided to make partial retrievals possible
+with simple pathing selectors [as described in this lotus changeset](https://github.com/filecoin-project/lotus/pull/6393#issue-661783290)
+
+## Limits in consideration
+- A car file accepted by the Filecoin network currently can have a maximum payload size of about 65,024MiB ( 64<<30 / 128 * 127 ). For the sake of argument let's assume a future with 128GiB sectors, which gives an exact maximum payload size of 136,365,211,648 bytes. Assuming ~60 bytes for a car header, **shortest safe CID** representation of a CIDv0 sha2-256 at 34 bytes, and payload of 1024 bytes per block ( ridiculously small NFTs ), gives us an upper bound of (136365211648 - 60) / ( 2 + 34 + 1024) ~~ 128,646,426 ~~ an upper bound of 2^27 individual CIDs that could be in a deal and all be individually addressable.
+
+- The **longest textual representation** of a "tenuously common" CID would be a `b`-multibase base32 representation of a [blake2b-512](https://cid.ipfs.io/#bafymbzacid777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777777), which clocks at 113 bytes. This in turn means that a typical UnixFS directory can contain:
+1048576 = ( 4b dir hdr ) + N * ( 2b pbframe hdr + 2b CID prefix + 70b CID + 2b name prefix + 113b text-CID + ~5 bytes for prefixed size ) ~~ 5405 such names without sharding, before going over the 1MiB libp2p limit. In order to be super-conservative assume a target of 2^12 entries per "aggregate shard". If we go with the more reasonable 256-bit hashes, we arrive at ~9363 names which translates to 2^13 entries per shard.
```

Comment:
What is the 1MiB libp2p limit?

Reply:
@alanshaw it's a simplification of "Libp2p will not relay any block over `(1<<21) - 1` bytes, so we cap everything within UnixFS land to half-that to be safe"